content/types/_realtime_client_options.textile

```diff
@@ -1,10 +1,14 @@
 
-- <span lang="default">recover</span><span lang="ruby">:recover</span> := This option allows a connection to inherit the state of a previous connection that may have existed under a different instance of the Realtime library. This might typically be used by clients of the browser library to ensure connection state can be preserved when the user refreshes the page. See "connection state recovery":/realtime/connection/#connection-state-recovery for further information on this option.<br>__Type: @String@__
+- <div lang="javascript">recover</div> := This option allows a connection to inherit the state of a previous connection that may have existed under a different instance of the Realtime library. This might typically be used by clients of the browser library to ensure connection state can be preserved when the user refreshes the page. A recovery key string can be explicitly provided, or alternatively if a callback function is provided, the client library will automatically persist the recovery key between page reloads and call the callback when the connection is recoverable. The callback is then responsible for confirming whether the connection should be recovered or not. See "connection state recovery":/realtime/connection/#connection-state-recovery for further information<br>__Type: @String@, @Callable@__
+
+- <div lang="javascript">closeOnUnload</div> := When true, the client library will automatically send a close request to Ably whenever the @window beforeunload@ event fires. By enabling this option, the close request sent to Ably ensures the connection state will not be retained and all channels associated with the channel will be detached. This is commonly used by developers who want presence leave events to fire immediately i.e. if a user navigates to another page or closes their browser, then enabling this option will result in the presence member leaving immediately. Without this option or an explicit call to the "@close@":/realtime/connection#close method of the "@Connection object@":/realtime/connection, Ably expects that the abruptly disconnected connection could later be recovered and therefore does not immediately remove the user from presence. Instead, to avoid "twitchy" presence behaviour an abruptly disconnected client is removed in channels they are present after 15 seconds, and the connection state is retained for two minutes<br>__Type: @Boolean@__
+
+- <div lang="java,ruby,nodejs,objc,swift"><span lang="default">recover</span><span lang="ruby">:recover</span></div> := This option allows a connection to inherit the state of a previous connection that may have existed under a different instance of the library. This might typically be used by clients of an app to ensure connection state can be preserved following a reload. See "connection state recovery":/realtime/connection/#connection-state-recovery for further information and example code<br>__Type: @String@__
```

Comment:
This could be more specific about how to use the option -- eg could add `by providing that connection's <span lang="default">"@recoveryKey@":#recovery-key</span><span lang="ruby">"@recovery_key@":#recovery-key</span>` at the end of the first sentence